### PR TITLE
docs: improve AI agent instructions for reliability

### PR DESCRIPTION
**Pull request by rjmurillo**
This PR updates the AI agent instructions to prevent recurring failures.

### Changes Made

- **Consolidated Rules:** Duplicated instructions for handling build failures have been consolidated into a single, authoritative section to prevent divergence.
- **Improved Clarity:** Added explicit rules for handling `AnalyzerReleases.md` files, prioritizing URLs in error messages, and ensuring the development workflow concludes with a pull request.

These changes are the result of a root cause analysis of recent failures and are intended to make the AI agent a more efficient and reliable contributor.

### CODE REVIEW

File: .cursor/rules/process-task-list.md [hunk 1]
```diff
@@ -19,10 +21,15 @@ Guidelines for managing task lists in GitHub issue Markdown files to track progr
       - **Formats the message as a single-line command using `-m` flags**, e.g.:
 
         ```text
-        git commit -m "feat: add payment validation logic" -m "- Validates card type and expiry" -m "- Adds unit tests for edge cases" -m "Related to #123 in Explainer"
+        git commit -m "feat: add payment validation logic" -m "- Validates card type and expiry" -m "- Adds unit tests for edge cases" -m "Related to #123 in Explainer" -m "Fixes sub-task #456"
         ```
 
-  3. Once all the subtasks are marked completed and changes have been committed, mark the **parent task** as completed.
+  3. Once all the subtasks are marked completed and changes have been committed, verify with the user the task is completed.
+  4. Once the user has indicated the work is verified, push the branch and open a pull request.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .cursor/rules/process-task-list.md:28]: The tasks are completed when the PR merges. This ensures a human has reviewed the work and not the AI itself. It's not good enough to be autonomous. A variation of this would be the Copilot model, placing the task list as a copy into the PR description, and then revising the PR description with additional tasks and progress referencing only the GitHub issue as the source/inspiration for the change (i.e., the "why" and the "what", rather than the GitHub PR being the "how").
